src/components/map/Map.jsx

```diff
@@ -1,42 +1,40 @@
-import React from 'react';
-import { compose, withProps, withStateHandlers, lifecycle } from 'recompose';
+import React, { Component } from 'react';
+import { compose, withProps, lifecycle } from 'recompose';
 import { withScriptjs, withGoogleMap, GoogleMap } from 'react-google-maps';
 import LocationMarker from './LocationMarker';
 import config from '../../config';
+import debounce from 'lodash/debounce';
```

Comment:
It makes little functional difference, but in terms of SoC, debouncing logic feels like it belongs more in the container that takes care of actions and state, rather than in the mostly-presentational component that manages the Google map.
(Also, as noted in another comment, that container will need debouncing for the search anyway.)

Is the reason for not putting it in the paren't onBoundsChanged the cost of calling computeDistanceBetween? I'd actually expect that would be a pretty trivial operation (as opposed to e.g. re-rendering).

Reply:
I think it can go in the MapView now.